File: MapData/backend/test_api.py
import requests
import json
import sys

# API endpoint URL
API_URL = "http://localhost:8000/api/v1/geographic/states"

def test_states_api():
    """Test the states API endpoint to verify it returns proper GeoJSON data."""
    print("Testing states API endpoint...")
    
    try:
        # Make a request to the API with include_geometry=true
        response = requests.get(API_URL, params={"include_geometry": True})
        
        # Check if the request was successful
        if response.status_code == 200:
            print(f"✅ API request successful (status code: {response.status_code})")
            
            # Parse the JSON response
            data = response.json()
            
            # Check if the response is a valid GeoJSON FeatureCollection
            if data.get("type") == "FeatureCollection" and "features" in data:
                print(f"✅ Response is a valid GeoJSON FeatureCollection with {len(data['features'])} features")
                
                # Check if we have features
                if len(data["features"]) > 0:
                    # Check the first feature
                    first_feature = data["features"][0]
                    
                    # Check if the feature has properties and geometry
                    if "properties" in first_feature and "geometry" in first_feature:
                        print("✅ Features have properties and geometry")
                        
                        # Check if area_sq_miles is in properties
                        if "area_sq_miles" in first_feature["properties"]:
                            area = first_feature["properties"]["area_sq_miles"]
                            name = first_feature["properties"]["name"]
                            print(f"✅ area_sq_miles found in properties: {name} = {area} sq miles")
                        else:
                            print("❌ area_sq_miles not found in properties")
                            
                        # Print a sample of the properties
                        print("\nSample properties from first feature:")
                        for key, value in first_feature["properties"].items():
                            print(f"  - {key}: {value}")
                    else:
                        print("❌ Features missing properties or geometry")
                else:
                    print("❌ No features found in the response")
            else:
                print("❌ Response is not a valid GeoJSON FeatureCollection")
                print(f"Actual response type: {data.get('type', 'unknown')}")
        else:
            print(f"❌ API request failed with status code: {response.status_code}")
            print(f"Response: {response.text}")
            
    except requests.exceptions.ConnectionError:
        print("❌ Connection error: Could not connect to the API server")
        print("Make sure the FastAPI server is running (uvicorn app:app --reload)")
    except Exception as e:
        print(f"❌ Error: {e}")

if __name__ == "__main__":
    test_states_api() 

File: MapData/frontend/src/test-api.js
import axios from 'axios';

// Define the API base URL
const API_BASE_URL = 'http://localhost:8000/api/v1';

// Test function to run in browser console
async function testApiConnection() {
  console.log('Testing API connection from frontend...');
  
  try {
    // Test health endpoint
    console.log('Testing health endpoint...');
    const healthResponse = await axios.get(`${API_BASE_URL}/geographic/health`);
    console.log('Health endpoint status:', healthResponse.status);
    console.log('Health endpoint data:', healthResponse.data);
    
    // Test states endpoint without geometry
    console.log('Testing states endpoint without geometry...');
    const statesResponse = await axios.get(`${API_BASE_URL}/geographic/states`, {
      params: { include_geometry: false }
    });
    console.log('States endpoint status:', statesResponse.status);
    console.log('States count:', statesResponse.data.features.length);
    
    // Check if states have area_sq_miles
    const sampleState = statesResponse.data.features[0];
    console.log('Sample state:', sampleState.properties.name);
    console.log('Sample state area:', sampleState.properties.area_sq_miles || 'Not found');
    
    // Test states endpoint with geometry
    console.log('Testing states endpoint with geometry...');
    const statesGeoResponse = await axios.get(`${API_BASE_URL}/geographic/states`, {
      params: { include_geometry: true }
    });
    console.log('States with geometry endpoint status:', statesGeoResponse.status);
    console.log('States with geometry count:', statesGeoResponse.data.features.length);
    
    // Check if states have geometry
    const sampleStateGeo = statesGeoResponse.data.features[0];
    const hasGeometry = sampleStateGeo.geometry.type !== 'Point' || (
      sampleStateGeo.geometry.coordinates[0] !== 0 && 
      sampleStateGeo.geometry.coordinates[1] !== 0
    );
    console.log('Sample state geometry type:', sampleStateGeo.geometry.type);
    console.log('Has valid geometry:', hasGeometry);
    
    return 'API tests completed successfully';
  } catch (error) {
    console.error('Error testing API:', error);
    if (error.response) {
      console.error('Error response data:', error.response.data);
      console.error('Error response status:', error.response.status);
    } else if (error.request) {
      console.error('No response received:', error.request);
    } else {
      console.error('Error setting up request:', error.message);
    }
    return 'API tests failed';
  }
}

// Export the test function to be used in the browser console
window.testApiConnection = testApiConnection;

export default testApiConnection; 